### PR TITLE
Move shared messaging library

### PR DESCRIPTION
**Pull request by jonasdmoeller**
## Description

Each domain has its own implementation of the messaging logic to convert protobuf message from and to internal formats. Code has been copy/pasted around the projects, but should be gathered one place instead.

## References

* #0000

### CODE REVIEW

File: source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+limitations under the License.
+-->
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj:23]: You can delete this line, and instead add the following group further below:
```
  <PropertyGroup>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
    <!-- Disable warning on your public types/methods for not having added full documentation tags -->
    <NoWarn>$(NoWarn);1591</NoWarn>
  </PropertyGroup>

```

File: source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    <AssemblyName>Messaging.Protobuf.Integration.ServiceCollection</AssemblyName>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj:24]: Should be "Energinet.DataHub.Core.Messaging.Protobuf.Integration.ServiceCollection"

File: source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    <RootNamespace>Energinet.DataHub.Core.Messaging.Protobuf</RootNamespace>
+    <LangVersion>9</LangVersion>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj:26]: I prefer to always be explicit and show which projects are "packable" by using: 
`<IsPackable>true</IsPackable>`

File: source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    <PackageDescription>
+      [Release Notes](https://github.com/Energinet-DataHub/geh-core/blob/master/source/Messaging/documents/release-notes/release-notes.md)
+      [Documentation](https://github.com/Energinet-DataHub/geh-core/blob/master/source/Messaging/documents/documentation.md)
+    </PackageDescription>
+    <Description>Helper library for dependency injection of Messaging Protobuf library.</Description>
+    <PackageTags>energinet;datahub;messaging;protobuf</PackageTags>
+    <PackageLicenseExpression>Apache-2.0</PackageLicenseExpression>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+  </PropertyGroup>
+
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj:57]: Support source link by adding:
```
  <!-- Enable Source Link (https://github.com/dotnet/sourcelink/blob/master/README.md) -->
  <PropertyGroup>
    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
    <PublishRepositoryUrl>true</PublishRepositoryUrl>
  </PropertyGroup>

```

File: source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  <ItemGroup>
+    <PackageReference Include="Google.Protobuf" Version="3.18.0" />
+    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.2" />
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf.Integration.ServiceCollection/Messaging.Protobuf.Integration.ServiceCollection.csproj:60]: Add to support source link:
```
    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>

```

File: source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
+    <LangVersion>9</LangVersion>
+    <AssemblyName>Energinet.DataHub.Core.Messaging.Protobuf</AssemblyName>
+    <RootNamespace>Energinet.DataHub.Core.Messaging.Protobuf</RootNamespace>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj:25]: I prefer to always be explicit and show which projects are "packable" by using: 
`<IsPackable>true</IsPackable>`

File: source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+    <!-- Disable warning on your public types/methods for not having added full documentation tags -->
+    <NoWarn>$(NoWarn);1591</NoWarn>
+  </PropertyGroup>
+
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj:63]: Add:
```
  <!-- Enable Source Link (https://github.com/dotnet/sourcelink/blob/master/README.md) -->
  <PropertyGroup>
    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
    <PublishRepositoryUrl>true</PublishRepositoryUrl>
  </PropertyGroup>
```

File: source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+  <ItemGroup>
+    <PackageReference Include="Google.Protobuf" Version="3.18.0" />
```
Comment (dstsyst) [inline source/Messaging/source/Messaging.Protobuf/Messaging.Protobuf.csproj:65]: Add:
```
    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>

```

File: source/Messaging/source/Messaging/Messaging.csproj [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <Nullable>enable</Nullable>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+    <RunAnalyzersDuringBuild>true</RunAnalyzersDuringBuild>
+    <AssemblyName>Energinet.DataHub.Core.Messaging.Integration.ServiceCollection</AssemblyName>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging/Messaging.csproj:22]: Should be "Energinet.DataHub.Core.Messaging"

Comment (dstsyst) [reply source/Messaging/source/Messaging/Messaging.csproj:22]: I prefer to always be explicit and show which projects are "packable" by using: 
`<IsPackable>true</IsPackable>`

File: source/Messaging/source/Messaging/Messaging.csproj [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    </PackageReleaseNotes>
+    <!-- PackageDescription:
+      Is shown in GitHub packages "About this package" section,
+      and in Visual Studio package manager view.
+    -->
+    <PackageDescription>
+      [Release Notes](https://github.com/Energinet-DataHub/geh-core/blob/master/source/Messaging/documents/release-notes/release-notes.md)
+      [Documentation](https://github.com/Energinet-DataHub/geh-core/blob/master/source/Messaging/documents/documentation.md)
+    </PackageDescription>
+    <Description>Library to support the message ingestion and processing of integration events in Green Energy Hub.</Description>
```
Comment (dstsyst) [inline source/Messaging/source/Messaging/Messaging.csproj:51]: As we have decided to use "Energinet.DataHub" for core libraries at the moment, I think that maybe we should not use "Green Energy Hub" but instead "Energinet DataHub" in the description.

File: source/Messaging/source/Messaging/Messaging.csproj [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <DocumentationFile>bin\$(Configuration)\$(TargetFramework)\$(AssemblyName).xml</DocumentationFile>
+    <!-- Disable warning on your public types/methods for not having added full documentation tags -->
+    <NoWarn>$(NoWarn);1591</NoWarn>
+  </PropertyGroup>
+
```
Comment (dstsyst) [inline source/Messaging/source/Messaging/Messaging.csproj:63]: Add:   
```
<!-- Enable Source Link (https://github.com/dotnet/sourcelink/blob/master/README.md) -->
  <PropertyGroup>
    <!-- Publish the repository URL in the built .nupkg (in the NuSpec <Repository> element) -->
    <PublishRepositoryUrl>true</PublishRepositoryUrl>
  </PropertyGroup>

```

File: source/Messaging/source/Messaging/Messaging.csproj [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+  <ItemGroup>
+    <PackageReference Include="FluentValidation" Version="9.5.4" />
+    <PackageReference Include="MediatR" Version="9.0.0" />
+    <PackageReference Include="NodaTime" Version="3.0.7" />
```
Comment (dstsyst) [inline source/Messaging/source/Messaging/Messaging.csproj:67]: Add:
```
    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>

```

**Review by dstsyst — COMMENTED** [review]
Added a few comments

**Review by dstsyst — APPROVED** [review]

**Review by ZavezX — APPROVED** [review]

**Pull request merged**